src/coreclr/vm/ecalllist.h

```diff
@@ -159,31 +159,30 @@ FCFuncStart(gCOMTypeHandleFuncs)
-    FCFuncElement("_GetClassLayout", MetaDataImport::GetClassLayout)
-    FCFuncElement("_GetSignatureFromToken", MetaDataImport::GetSignatureFromToken)
-    FCFuncElement("_GetNamespace", MetaDataImport::GetNamespace)
-    FCFuncElement("_GetEventProps", MetaDataImport::GetEventProps)
-    FCFuncElement("_GetFieldDefProps", MetaDataImport::GetFieldDefProps)
-    FCFuncElement("_GetPropertyProps", MetaDataImport::GetPropertyProps)
-    FCFuncElement("_GetParentToken", MetaDataImport::GetParentToken)
-    FCFuncElement("_GetParamDefProps", MetaDataImport::GetParamDefProps)
-    FCFuncElement("_GetGenericParamProps", MetaDataImport::GetGenericParamProps)
-
-    FCFuncElement("_Enum", MetaDataImport::Enum)
-    FCFuncElement("_GetMemberRefProps", MetaDataImport::GetMemberRefProps)
-    FCFuncElement("_GetCustomAttributeProps", MetaDataImport::GetCustomAttributeProps)
-    FCFuncElement("_GetFieldOffset", MetaDataImport::GetFieldOffset)
-
-    FCFuncElement("_GetSigOfFieldDef", MetaDataImport::GetSigOfFieldDef)
-    FCFuncElement("_GetSigOfMethodDef", MetaDataImport::GetSigOfMethodDef)
-    FCFuncElement("_GetFieldMarshal", MetaDataImport::GetFieldMarshal)
-    FCFuncElement("_GetPInvokeMap", MetaDataImport::GetPinvokeMap)
-    FCFuncElement("_IsValidToken", MetaDataImport::IsValidToken)
-    FCFuncElement("_GetMarshalAs", MetaDataImport::GetMarshalAs)
+    FCFuncElement("GetDefaultValue", MetaDataImport::GetDefaultValue)
+    FCFuncElement("GetName", MetaDataImport::GetName)
+    FCFuncElement("GetUserString", MetaDataImport::GetUserString)
+    FCFuncElement("GetScopeProps", MetaDataImport::GetScopeProps)
+    FCFuncElement("GetClassLayout", MetaDataImport::GetClassLayout)
+    FCFuncElement("GetSignatureFromToken", MetaDataImport::GetSignatureFromToken)
+    FCFuncElement("GetNamespace", MetaDataImport::GetNamespace)
+    FCFuncElement("GetEventProps", MetaDataImport::GetEventProps)
+    FCFuncElement("GetFieldDefProps", MetaDataImport::GetFieldDefProps)
+    FCFuncElement("GetPropertyProps", MetaDataImport::GetPropertyProps)
+    FCFuncElement("GetParentToken", MetaDataImport::GetParentToken)
+    FCFuncElement("GetParamDefProps", MetaDataImport::GetParamDefProps)
+    FCFuncElement("GetGenericParamProps", MetaDataImport::GetGenericParamProps)
+
+    FCFuncElement("GetMemberRefProps", MetaDataImport::GetMemberRefProps)
+    FCFuncElement("GetCustomAttributeProps", MetaDataImport::GetCustomAttributeProps)
+    FCFuncElement("GetFieldOffset", MetaDataImport::GetFieldOffset)
+
+    FCFuncElement("GetSigOfFieldDef", MetaDataImport::GetSigOfFieldDef)
+    FCFuncElement("GetSigOfMethodDef", MetaDataImport::GetSigOfMethodDef)
+    FCFuncElement("GetFieldMarshal", MetaDataImport::GetFieldMarshal)
+    FCFuncElement("GetPInvokeMap", MetaDataImport::GetPInvokeMap)
+    FCFuncElement("IsValidToken", MetaDataImport::IsValidToken)
+    FCFuncElement("GetMarshalAs", MetaDataImport::GetMarshalAs)
```

Comment:
Q. Is converting FCall->QCall these contracts at this point regressing (not straightforward and require extra work etc.)? I realize that removing `FCThrow` is a priority (to reduce the reliance on libunwind among other things), but still replacing the entire FCall to QCall/managed, specifically the ones which we are touching as part of this work, feels like a general goodness.

Reply:
The only one that was converted to a QCall was the `_Enum`, which didn't show any obvious regression. However, changing all of these, some very lightweight, is likely to introduce a regression since the transition itself could impact the call. There is nothing wrong in general about FCalls so leaving these as is for now is appropriate. We will always have some FCalls in the runtime though.

Reply:
> There is nothing wrong in general about FCalls

Per https://github.com/dotnet/runtime/blob/f6d3d4c80f45029ae41ca50eed8e754156edb6c6/docs/design/coreclr/botr/corelib.md#choosing-between-fcall-qcall-pinvoke-and-writing-in-managed-code, FCalls are referred to as  relics of past, so my take away is new FCall are less inspiring in general. :)

Reply:
Yes, they are. However, if one reads the full section there are two points to take away.

1) These are not new, they are existing that are removing Helper Method Frames - the intent of the PR.

> QCalls are the preferred mechanism going forward. You should only use FCalls when you are "forced" to. This happens when there is common "short path" through the code that is important to optimize.

2) The reflection code paths are just the sort of "short path" mentioned in the alluded to in the above statement that are important to optimize.

I'm not saying we shouldn't endevour to remove QCalls, but these are not appropriate to convert at this point.

Reply:
(I was just commenting on the performance aspect. I do not have a strong opinion about whether or not the FCalls should be converted to QCalls in this PR.)